trpcultivate_phenoshare/src/Plugin/TripalImporter/TripalCultivatePhenoshareImporter.php

```diff
@@ -216,12 +229,52 @@ public function stage1(&$form, $form_state, $stage_status = '') {
     $form[ $fld_wrapper ]['project'] = [
       '#title' => t('Project/Experiment'),
       '#type' => 'textfield',
-      '#description' => t('Type in the experiment or project title your data is specific to.'),
-      '#weight' => -100,      
+      '#weight' => -100,  
+      '#required' => TRUE,   
+      '#description' => t('Type in the experiment or project title your data is specific to.'), 
       '#attributes' => ['placeholder' => 'Project/Experiment Name', 'class' => ['tcp-autocomplete']],
       '#autocomplete_route_name' => 'tripal_chado.project_autocomplete',
       '#autocomplete_route_parameters' => ['type_id' => 0, 'count' => 5],
+      
+      // Used by script to pre-select genus paired to project entered.
+      '#id' => 'trpcultivate-fld-project'
     ];
+    
+    // Field Genus:
+    // Prepare select options with only active genus.
+    $all_genus = $this->connection->query("SELECT genus FROM {1:organism}");
+    
+    // Array to hold all active genus.
+    $active_genus = [];
+    foreach($all_genus as $obj_genus) {
+      $genus = $obj_genus->genus;
+
+      $genus_config = $this->service_genusontology->getGenusOntologyConfigValues($genus);
+      if ($genus_config && $genus_config['trait']) {
+        $active_genus[ $genus ] = $genus;
+      }
+    } 
+
+    asort($active_genus);
```

Comment:
Lines 243-258 might be best moved into an additional method in the genus ontology service. For example,

```php
/**
 * Retrieve a sorted list of all configured genus.
 * 
 * @return array
 *   Array where the key and value are the genus.
 */
public function getConfiguredGenusList();
```

Reply:
applied, this is good 👍